Code/Framework/AzCore/AzCore/Settings/SettingsRegistryOriginTracker.h

```diff
@@ -0,0 +1,113 @@
+        void SetTrackingFilter(TrackingFilterCallback filterCallback);
+
+        //! Gets setting registry being tracked
+        AZ::SettingsRegistryInterface& GetSettingsRegistry();
+        const AZ::SettingsRegistryInterface& GetSettingsRegistry() const;
+
+        //! Returns the last origin associated with the given key
+        bool FindLastOrigin(AZ::IO::Path& originPath, AZStd::string_view key) const;
+
+        struct SettingsRegistryOrigin
+        {
+            //! Path to the file where the settings is sourced from or a special value in angle brackets to indicate a non-file source(<C++>, <command-line>, etc...)
+            AZ::IO::Path m_originFilePath;
+            //! The value of the settings key associated with the origin
+            AZStd::string m_settingsKey;
+
+            //! Sutable for representing a setting as a JSON string or to be not engaged if the setting has been deleted.
+            using ValueType = AZStd::optional<AZStd::string>;
+
+            //! The value of the setting as a JSON string at the time the origin entry was created
+            //! If the value is a nullopt, then the setting was removed
+            ValueType m_settingsValue;
+
+            bool operator==(const SettingsRegistryOrigin& origin) const;
+        };
+
+        using SettingsRegistryOriginStack = AZStd::deque<SettingsRegistryOrigin>;
+
+        //! Origin Visitor is callback is invoked for each setting key found during the VisitOrigins method
+        //! @param settingsRegistryOrigin entry being visited
+        //! @returns True should be returned if the visitation should continue.
+        using OriginVisitorCallback = AZStd::function<bool(const SettingsRegistryOrigin&)>;
+
+        //! Visit all settings keys that are at specified key path and any children in the DomPrefixTree
+        //! and invoke the supplied callback with the visit key and the SettingsRegistryOriginStack
+        void VisitOrigins(AZStd::string_view key, const OriginVisitorCallback& visitCallback) const;
+
+    private:
+
+        struct SettingsNotificationHandler
+        {
+            SettingsNotificationHandler(SettingsRegistryOriginTracker& originTracker);
+            ~SettingsNotificationHandler();
+
+            void operator()(const AZ::SettingsRegistryInterface::NotifyEventArgs& notifyEventArgs);
+
+        private:
+            SettingsRegistryOriginTracker& m_originTracker;
+            AZ::JsonSerializationResult::JsonIssueCallback m_prevReportingCallback;
+        };
+
+        using SettingsRegistryOriginPrefixTree = AZ::Dom::DomPrefixTree<SettingsRegistryOriginStack>;
+
+        //! Origin Visitor is callback is invoked for each setting key found during the VisitOrigins method
+        //! @param settingsRegistryOrigin entry being visited
+        //! @returns True should be returned if the visitation should continue.
+        using OriginVisitorCallback = AZStd::function<bool(const SettingsRegistryOrigin&)>;
```

Comment:
The OriginVisitorCallback is aliased listed twice, causing a compilation issue on Linux https://jenkins.build.o3de.org/job/O3DE/job/PR-11028/1/execution/node/360/log/

Since this has been defined up above on line 69, it can be deleted here.